core/src/main/java/org/infinispan/util/concurrent/locks/OwnableRefCountingReentrantLock.java

```diff
@@ -13,15 +13,24 @@
  * @see OwnableReentrantPerEntryLockContainer
  */
 public class OwnableRefCountingReentrantLock extends OwnableReentrantLock implements RefCountingLock {
-   private final AtomicInteger references = new AtomicInteger(1);
+   private final static AtomicIntegerFieldUpdater<OwnableRefCountingReentrantLock> updater
```

Comment:
Actually, Stale Pedersen has reminded me of this one :) I was almost about to put the AtomicInteger magic directly into this class :)